integreat_compass/cms/views/offers/my_offers_list_view.py

```diff
@@ -0,0 +1,84 @@
+import logging
+
+from django.contrib import messages
+from django.db import transaction
+from django.shortcuts import redirect, render
+from django.utils.decorators import method_decorator
+from django.utils.translation import gettext as _
+from django.views.generic import TemplateView
+
+from ...decorators import permission_required
+from ...models import Offer
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(permission_required("cms.change_offer"), name="dispatch")
+class MyOffersListView(TemplateView):
+    """
+    View showing all of a user's offers
+    """
+
+    template_name = "offers/my_offers_list.html"
+
+    def get(self, request, *args, **kwargs):
+        r"""
+        Render list of a user's offers.
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+
+        offers = Offer.objects.filter(creator=request.user)
+        return render(
+            request,
+            self.template_name,
+            {**self.get_context_data(**kwargs), "offers": offers},
+        )
+
+    @transaction.atomic
+    def post(self, request, *args, **kwargs):
```

Comment:
I think it makes sense to outsource the deletion into a separate view in case you want to allow deletion from other views as well.
You could e.g. create a new view `OfferDeleteView` which inherits from [`DeleteView`](https://docs.djangoproject.com/en/4.2/ref/class-based-views/generic-editing/#deleteview) and then you only have to override `form_valid` to show the success message.

And if you want to make sure a user can only delete their own offers, it would make more sense to also add [`PermissionRequiredMixin`](https://docs.djangoproject.com/en/4.2/topics/auth/default/#the-permissionrequiredmixin-mixin) as second base class and overwrite `get_permission_required()` to make sure to raise a 403 error if a user tries to delete a different offer than their own...